apps/hash-frontend/src/pages/[shortname]/types/entity-type/[...slug-maybe-version].page/definition-tab/use-editor-ontology-functions.ts

```diff
@@ -123,11 +126,26 @@ export const useEditorOntologyFunctions = (
     [generateTypeUrlsForUser, getEntityType, getPropertyType],
   );
 
+  const canEditResource = useCallback(
+    (resourceNamespace: string) => {
+      if (!resourceNamespace || !user) {
+        return false;
+      }
+
+      return !!(
+        resourceNamespace === user.shortname ||
+        user.memberOf.find((org) => org.shortname === resourceNamespace)
+      );
```

Comment:
There's an existing function in the codebase `canUserEditResource` which we could use in place of this.

Reply:
Thanks for the suggestion, I didn't use that at first because I didn't really have access to the accountId of the property types and link types, but @Alfred-Mountfield pointed out that checking against the namespace was flawed using the previous solution for types with the same namespace and different domains (`blockprocol.org/@bob01/types/entity-type/test and hash.ai/@bob01/types/entity-type/test would have the same namespace, even though the domain is different).

So I had to make changes to `LatestPropertyTypesContextProvider` and `useEntityTypeValue` to also get the types' metadata and pass it to `useEditorOntologyFunctions` to then check against the resource that `canEditResource` was called with.